probe-rs/src/flashing/flasher.rs

```diff
@@ -481,6 +691,15 @@ impl Flasher {
         ignore_filled: bool,
     ) -> Result<bool, FlashError> {
         let encoding = self.flash_algorithm.transfer_encoding;
+
+        // Check if CRC32 verification is available first (preferred)
+        if self.flash_algorithm.pc_crc32.is_some() {
```

Comment:
Because CRC can have collisions, it would be nice to have a way to opt out of this via the CLI options, just in case.

Reply:
True, that's a reasonable point. Would you prefer something like an additional flasg eg. `--preverify --no-crc` or a separate flag `--preverify-read` or similar?